src/test/rpc/TransactionEntry_test.cpp

```diff
@@ -143,29 +144,43 @@ class TransactionEntry_test : public beast::unit_test::suite
     }
 
     void
-    testRequest()
+    testRequest(unsigned apiVersion)
     {
-        testcase("Basic request");
+        testcase("Basic request API version " + std::to_string(apiVersion));
         using namespace test::jtx;
         Env env{*this};
 
-        auto check_tx = [this, &env](
+        auto check_tx = [this, &env, apiVersion](
                             int index,
                             std::string const txhash,
-                            std::string const expected_json = "") {
+                            std::string const expected_json = "",
+                            std::string const close_time_iso = "") {
             // first request using ledger_index to lookup
-            Json::Value const resIndex{[&env, index, &txhash]() {
+            Json::Value const resIndex{[&env, index, &txhash, apiVersion]() {
                 Json::Value params{Json::objectValue};
                 params[jss::ledger_index] = index;
                 params[jss::tx_hash] = txhash;
+                params[jss::api_version] = apiVersion;
                 return env.client().invoke(
                     "transaction_entry", params)[jss::result];
             }()};
 
             if (!BEAST_EXPECTS(resIndex.isMember(jss::tx_json), txhash))
                 return;
 
-            BEAST_EXPECT(resIndex[jss::tx_json][jss::hash] == txhash);
+            if (apiVersion > 1)
+            {
+                BEAST_EXPECT(resIndex[jss::hash] == txhash);
+                BEAST_EXPECT(resIndex[jss::validated] == true);
+                BEAST_EXPECT(!resIndex[jss::tx_json].isMember(jss::Amount));
+
+                if (BEAST_EXPECT(!close_time_iso.empty()))
+                    BEAST_EXPECT(
+                        resIndex[jss::close_time_iso] == close_time_iso);
+            }
+            else
+                BEAST_EXPECT(resIndex[jss::tx_json][jss::hash] == txhash);
```

Comment:
Could you add checks for the same fields, or their expected absence, as in the `apiVersion > 1` block?

Reply:
Thanks, done.

Reply:
Also, thanks to this feedback from you, I found and fixed (harmless, but still) bug in `STTx::getJson` - we were populating `hash` inside transaction output from API version 2, if overload without `bool binary` was used.